_build.ham

```diff
@@ -0,0 +1,16 @@
+SubDir TOP ;
+
+local BIN_DIR = [ FDirName $(TOP) bin ] ;
+# ECHO "... BIN_DIR:" $(BIN_DIR) ;
+
+# Recursively get all .sh files inside of bin/ directory
+local SH_FILES = [ FGristFiles [ ListFilesR $(BIN_DIR) : *.sh ] ] ;
+# Add all files at top level of bin/ directory with no file suffix
+SH_FILES += [ FGristFiles [ Match (^[^.]*$) : [ ListFiles $(BIN_DIR) ] ] ] ;
```

Comment:
So shfmt actually has a flag to return a list of bash files, I believe it actually reads the headers. I'm wondering if the best way is to not sure list them explicitly and just add a comment/rule that says "if you had a script without .sh you must add it here aswell". The initial list we can build by hand by running shfmt to list the files.

We can call a process in the ham file (I think the Bash or Shell rule?)... but I dont like it, its generally slow and you pay the price when parsing the ham file which is really bad for build performance & maintenance.